### ISSUE TITLE
[FEATURE REQUEST] Add multi-threading/processing to the Dataset Validator

### ISSUE THREAD

**Issue by chavinlo** [harubaru/waifu-diffusion#52]
It's awfully slow.

I am willing to do it if someone gives me directions

**Comment by chinoll** [comment]
Did you install xformers?

**Comment by chavinlo** [comment]
> Did you install xformers? [huggingface/diffusers#1343](https://github.com/huggingface/diffusers/issues/1343)

yes but I don't think xformers has anything to do with the validation process...

**Comment by cafeai** [comment]
You can skip validation, which is what I would suggest. Internally, I'm personally using a preprocessing application written in Rust. This probably isn't something you want to do in Python.

**Comment by lopho** [comment]
small self plug for a preprocessor written in python and fully parallel:
https://github.com/lopho/parallel_dataprocessor

**Comment by lopho** [comment]
#59 has parallel validation and migration
